test/integration/abuseReportTest.ts

```diff
@@ -58,85 +50,73 @@ describe("Test: Reporting abuse", async () => {
-            throw e;
-        }
-
-        try {
-            await goodUser.doRequest("POST", `/_matrix/client/r0/rooms/${encodeURIComponent(roomId)}/report/${encodeURIComponent(badEventId3)}`, "");
-            reportsToFind.push({
+            },
+            {
                 reporterId: goodUserId,
                 accusedId: badUserId,
                 eventId: badEventId3,
                 text: badText3,
                 comment: null,
-            });
-        } catch (e) {
-            console.error("Could not send third report", e.body || e);
-            throw e;
-        }
-
-        try {
-            await goodUser.doRequest("POST", `/_matrix/client/r0/rooms/${encodeURIComponent(roomId)}/report/${encodeURIComponent(badEventId4)}`, "");
-            reportsToFind.push({
+            },
+            {
                 reporterId: goodUserId,
                 accusedId: badUserId,
                 eventId: badEventId4,
                 text: null,
                 textPrefix: badText4.substring(0, 256),
                 comment: null,
-            });
-        } catch (e) {
-            console.error("Could not send fourth report", e.body || e);
-            throw e;
-        }
-
-        try {
-            await goodUser.doRequest("POST", `/_matrix/client/r0/rooms/${encodeURIComponent(roomId)}/report/${encodeURIComponent(badEventId5)}`, "");
-            reportsToFind.push({
+            },
+            {
                 reporterId: goodUserId,
                 accusedId: badUserId,
                 eventId: badEventId5,
                 text: null,
                 textPrefix: badText5.substring(0, 256).split("\n").join(" "),
                 comment: null,
-            });
-        } catch (e) {
-            console.error("Could not send fifth report", e.body || e);
-            throw e;
+            }
+        ]
+ 
+        const sendReport = async (report: any) => {
+            try {
+                await goodUser.doRequest("POST", `/_matrix/client/r0/rooms/${encodeURIComponent(roomId)}/report/${encodeURIComponent(report.eventId)}`,
+                    null,
+                    report.comment ? { reason: report.comment } : null
+                );
+                reportsToFind.push(report);
+            } catch (e) {
+                console.error("Could not send a report", e.body || e);
+                throw e;
+            }
         }
 
-        console.log("Test: Reporting abuse - wait");
-        await new Promise(resolve => setTimeout(resolve, 1000));
+        console.log("Test: Reporting abuse - sending each report and waiting for them to be received");
+        let notices = [];
+        await new Promise(resolve => {
+            matrixClient().on("room.event", (roomId, event) => {
+                if (roomId = this.mjolnir.managementRoomId) {
+                    notices.push(event);
+                }
+                if (notices.length === reportsToFind.length) {
```

Comment:
I'd be more comfortable if we made this a set or a map and removed reports as they arrive.